Add --rewrite flag to get command

jeroenvervaeke opened the pull request:
I'm working on a project that uses two versions of the same package.
- One version is the most recent version of the package (example: `github.com/mypackage/v119`)
- One version is an older version that we keep pinned to a specific version because the new one contains a bug that will not be fixed in the foreseeable future. (example: `github.com/mypackage/v100`)

With the current version of this tool, when you run `gomod get github.com/mypackage/v120`, it'll update both the most recent version (`github.com/mypackage/v119`) and the pinned/frozen version (`github.com/mypackage/v100`).
I only want to update `github.com/mypackage/v119` and leave `github.com/mypackage/v100` intact.

I've implemented the `--exact [package]` feature for this.

How to use this new flag:
```
gomod get --exact github.com/mypackage/v119 github.com/mypackage/v120
```

Behaviour:
Only `github.com/mypackage/v119` imports will be updated, and `github.com/mypackage/v100` imports stay intact.

icholy commented:
* Instead of the `Exact` field, change the `OnRewrite` signature to return an `error`. Return `imporpaths.ErrSkip` in the callback to prevent rewriting the specific import.
* Rename the parameter to `--rewrite` and have it take a regular expression to filter the imports to rewrite.

jeroenvervaeke commented:
Thanks for your time, I've addressed the comments.
- `OnRewrite` returns an `error`
- In the `OnRewrite` for a single package I've added the logic to check if the correct package is updated, return `ErrSkip` if not
- Renamed `--exact` to `--rewrite`
- `--rewrite` now accepts a regex instead of a string

icholy commented on main.go:
```diff
@@ -117,14 +118,14 @@ func listcmd(args []string) error {
 	fset.StringVar(&dir, "dir", ".", "working directory")
 	fset.BoolVar(&cached, "cached", true, "only fetch cached content from the module proxy")
-	fset.StringVar(&exact, "exact", "", "exact package version to upgrade")
+	fset.StringVar(&rewrite, "rewrite", "", "exact package version to upgrade")
```
```suggestion
	fset.TextVar(&rewrite, "rewrite", regexp.MustCompile(".*"), "exact package version to upgrade")
```

icholy commented on main.go:
```diff
@@ -117,13 +118,14 @@ func listcmd(args []string) error {
 
 func getcmd(args []string) error {
-	var dir string
+	var dir, rewrite string
```
```suggestion
	var rewrite regexp.Regexp
	var dir, string
```

icholy commented:
Thanks for the contribution!

icholy commented:
Tagged as v0.14.0

jeroenvervaeke commented:
Thanks for the fast reply!
